src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -713,6 +769,38 @@
     end
   end
 
+  describe 'POST #save_meta' do
+    before do
+      login user
+    end
+
+    context 'with an nonexistent project' do
+      let(:post_save_meta) { post :save_meta, params: { project: 'nonexistent_project' }, xhr: true }
+
+      it { expect{ post_save_meta }.to raise_error(Pundit::NotDefinedError) }
```

Comment:
I understand that it's throwing this exception but is that really what we expect? To me it sounds like `Pundit::NotDefinedError` is  not what we want to throw here.